src/Forms/CheckLabel.tsx

```diff
@@ -0,0 +1,134 @@
+import React, {
+  ReactElement,
+  useContext,
+  FunctionComponent,
+} from 'react';
+import styled, { ThemeContext } from 'styled-components';
+import { fromTheme } from '../Theme';
+import { POSITION, RequiredSymbol } from './InputLabel';
+
+export interface StyledCheckLabelProps {
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** Specifies the id for the label */
+  htmlFor: string;
+}
+
+export interface StyledCheckLabelTextProps {
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** Used to style label text in a different style if disabled is true */
+  disabled?: boolean;
+}
+
+export interface LabelProps {
+  /** The id of the field tied to this label */
+  htmlFor: string;
+  /** Specifies the label text */
+  label: string;
+  /** Allows you to pass in a label position property from the POSITION enum */
+  labelPosition?: POSITION;
+  /** If true, label will be visible */
+  isLabelVisible?: boolean;
+  /** If true, the label will be styled to indicate that it labels a required field */
+  isRequired?: boolean;
+  /** Used to style label text in a different style if disabled is true */
+  disabled?: boolean;
+}
+
+const generateGrid = (
+  labelPosition: POSITION,
+  isLabelVisible: boolean
+): string => {
+  if (!isLabelVisible) {
+    return `"input input input"
+            "e e e"`;
+  }
+  if (labelPosition === POSITION.RIGHT) {
+    return `"input label label"
+            ". e e"`;
+  }
+  return `"label input input"
+          ". e e"`;
+};
+
+const StyledCheckLabel = styled.label<StyledCheckLabelProps>`
+  display: grid;
+  grid-template-columns: ${({ labelPosition }) => (
+    labelPosition === POSITION.RIGHT
+      ? 'auto minmax(0, 1fr)'
+      : 'repeat(3, 1fr)'
+  )};
+  grid-template-rows: 1fr minmax(1em, max-content);
+  grid-template-areas: ${({ labelPosition, isLabelVisible }) => (
+    generateGrid(labelPosition, isLabelVisible)
+  )};
+  margin: ${fromTheme('ws', 'small')};
+  align-items: baseline;
+  gap: ${({ theme }) => (theme.ws.xsmall) + ' ' + (theme.ws.xsmall)};
+`;
+
+const StyledCheckLabelText = styled.span<StyledCheckLabelTextProps>`
+  display: ${({ isLabelVisible }) => (isLabelVisible ? 'inline' : 'none')};
+  grid-area: label;
+  justify-self: ${({ labelPosition }) => (
+    (labelPosition === POSITION.TOP || labelPosition === POSITION.RIGHT)
```

Comment:
As noted above, we can probably leave `POSITION.TOP` out of the Checkboxes

Reply:
This was also updated in 673b37e (above).